src/gj.rs

```diff
@@ -885,12 +909,20 @@ impl<'a> std::fmt::Display for TrieAccess<'a> {
 }
 
 impl<'a> TrieAccess<'a> {
-    fn filter_live<'b: 'a>(&'b self, ixs: &'b [Offset]) -> impl Iterator<Item = usize> + 'a {
-        ixs.iter().copied().filter_map(|ix| {
+    fn filter_live<'b: 'a>(
+        &'b self,
+        ixs: &'b [Offset],
+        include_subsumed: bool,
+    ) -> impl Iterator<Item = usize> + 'a {
+        ixs.iter().copied().filter_map(move |ix| {
             let ix = ix as usize;
             let (inp, out) = self.function.nodes.get_index(ix)?;
             if self.timestamp_range.contains(&out.timestamp)
                 && self.constraints.iter().all(|c| c.check(inp, out))
+                // If we are querying to run a rule, we should not include subsumed expressions
+                // but if we are querying to run a check we can include them.
+                // So we have a flag to control this behavior and pass it down to here.
+                && (include_subsumed || !self.function.nodes.get_index_row(ix).unwrap().subsumed)
```

Comment:
Unfortunately, this was the cleanest way I found to make the query engine ignore the subsumed values, only when we were running rules, but not when running a check.

Reply:
I agree that passing include_subsumed everywhere is not ideal- but it's okay, I can't think of another way either.
If we need to have more metadata in the future we can package it up.

Reply:
`include_subsumed` seemed to be an intrinsic property of `TrieAccess`, just like `timestamp_range`. It defines whether TrieAccess should build a trie only based on unsubsumed rows. So ideally, it should only be passed at places where `timestamp_range` is also passed. Can't we make include_subsumed a field of TrieAccess instead of passing it everywhere?